client/components/Windows/SearchResults.vue

```diff
@@ -0,0 +1,192 @@
+<template>
+	<div id="chat-container" class="window">
+		<div
+			id="chat"
+			:class="{
+				'colored-nicks': $store.state.settings.coloredNicks,
+				'show-seconds': $store.state.settings.showSeconds,
+			}"
+		>
+			<div
+				class="chat-view"
+				data-type="search-results"
+				aria-label="Search results"
+				role="tabpanel"
+			>
+				<div class="header">
+					<SidebarToggle />
+					<span class="title"
+						>Search results for "{{ $route.params.term }}" in
+						{{ $route.params.target }}</span
+					>
+					<span class="topic"></span>
+					<MessageSearchForm :network="network" :channel="channel" />
+				</div>
+				<div class="chat-content">
+					<div ref="chat" class="chat" tabindex="-1">
+						<div v-show="moreResultsAvailable" class="show-more">
+							<button
+								ref="loadMoreButton"
+								:disabled="
+									$store.state.messageSearchInProgress ||
+										!$store.state.isConnected
+								"
+								class="btn"
+								@click="onShowMoreClick"
+							>
+								<span v-if="$store.state.messageSearchInProgress">Loading…</span>
+								<span v-else>Show older messages</span>
+							</button>
+						</div>
+
+						<div v-if="$store.state.messageSearchInProgress" class="search-status">
+							Searching...
+						</div>
+						<div v-else-if="!messages.length" class="search-status">
+							No results found.
+						</div>
+						<div
+							v-else
+							class="messages"
+							role="log"
+							aria-live="polite"
+							aria-relevant="additions"
+						>
+							<template v-for="(message, id) in messages">
+								<DateMarker
+									v-if="shouldDisplayDateMarker(message, id)"
+									:key="message.time + '-date'"
+									:message="message"
+								/>
+								<Message
+									:key="message.time"
```

Comment:
I had this happen:
> [Vue warn]: Duplicate keys detected: '2019-12-27T18:15:45.232Z'. This may cause an update error.

Reply:
Also, context menus need fixing as they rely on `activeChannel` and that breaks context menus in search:

```
vue.js:105 TypeError: Cannot read property 'network' of null
    at VueComponent.openUserContextMenu 
```

Reply:
I'm getting duplicate key errors just by running a single search. There's no guarantee that the dates are unique and I indeed have two messages with the exact same date in my history. I don't know what to do about that since we don't have unique message IDs :(